diff --git a/Classes/ViewHelpers/ImageViewHelper.php b/Classes/ViewHelpers/ImageViewHelper.php
index 2c609cd..a513d64 100644
--- a/Classes/ViewHelpers/ImageViewHelper.php
+++ b/Classes/ViewHelpers/ImageViewHelper.php
@@ -179,6 +179,7 @@ class ImageViewHelper extends AbstractTagBasedViewHelper
                     'maxHeight' => $this->arguments['maxHeight'],
                     'crop' => $cropArea->isEmpty() ? null : $cropArea->makeAbsoluteBasedOnFile($image),
                 ];
+                $processingInstructions['rotate'] = $cropVariantCollection->getRotate($cropVariant);
                 if (!empty($this->arguments['fileExtension'] ?? '')) {
                     $processingInstructions['fileExtension'] = $this->arguments['fileExtension'];
                 }
diff --git a/Classes/ViewHelpers/MediaViewHelper.php b/Classes/ViewHelpers/MediaViewHelper.php
index 48e41dd..89e9136 100644
--- a/Classes/ViewHelpers/MediaViewHelper.php
+++ b/Classes/ViewHelpers/MediaViewHelper.php
@@ -146,6 +146,7 @@ class MediaViewHelper extends AbstractTagBasedViewHelper
             $processingInstructions['fileExtension'] = $fileExtension;
         }
         $imageService = $this->getImageService();
+        $processingInstructions['rotate'] = $cropVariantCollection->getRotate($cropVariant);
         $processedImage = $imageService->applyProcessingInstructions($image, $processingInstructions);
         $imageUri = $imageService->getImageUri($processedImage);
 
